src/mongoChangeStream.ts

```diff
@@ -272,7 +272,8 @@ export function initSync<ExtendedEvents extends EventEmitter.ValidEventTypes>(
           fullDocument: doc,
           operationType: 'insert',
           ns,
-        } as unknown as ChangeStreamInsertDocument
+          documentKey: { _id: doc?._id },
```

Comment:
It seems like `doc` shouldn't be undefined here, because if it is, then we should have broken out of the loop. Maybe `doc!._id` would be more appropriate?

Reply:
That's probably a good call. You can add a pipeline to the initial scan which makes what's in `doc` unpredictable, but you should always get an object.